src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -798,14 +904,212 @@ private void PlayWithCurrentSettingPressed()
+
+        public readonly List<Dictionary<int, PatchSnapshot>> PatchHistoryList;
+
+        /// <summary>
+        ///   This list stores all auto-evo results.
+        /// </summary>
+        public readonly List<LocalizedStringBuilder> RunResultsList;
+
+        /// <summary>
+        ///   Used to generate world statistics
+        /// </summary>
+        public readonly Dictionary<OrganelleDefinition, (double Percentage, double Average)>
+            MicrobeSpeciesOrganelleStatistics;
+
+        /// <summary>
+        ///   The current generation auto-evo has evolved
+        /// </summary>
+        public int CurrentGeneration;
+
+        /// <summary>
+        ///   Total used auto-evo time
+        /// </summary>
+        public TimeSpan TotalTimeUsed;
+
+        public AutoEvoExploringToolWorld(IAutoEvoConfiguration configuration)
+        {
+            AutoEvoConfiguration = configuration.Clone();
+            GameProperties = GameProperties.StartNewMicrobeGame(new WorldGenerationSettings
+                { AutoEvoConfiguration = AutoEvoConfiguration });
+            SpeciesHistoryList = new List<Dictionary<uint, Species>>();
+            PatchHistoryList = new List<Dictionary<int, PatchSnapshot>>();
+            RunResultsList = new List<LocalizedStringBuilder>();
+            CurrentGeneration = 0;
+            TotalTimeUsed = TimeSpan.Zero;
+
+            RunResultsList.Add(new LocalizedStringBuilder());
+            SpeciesHistoryList.Add(new Dictionary<uint, Species>
+            {
+                {
+                    GameProperties.GameWorld.PlayerSpecies.ID,
+                    (Species)GameProperties.GameWorld.PlayerSpecies.Clone()
+                },
+            });
+
+            PatchHistoryList.Add(GameProperties.GameWorld.Map.Patches.ToDictionary(pair => pair.Key,
+                pair => (PatchSnapshot)pair.Value.CurrentSnapshot.Clone()));
+
+            PatchesCount = GameProperties.GameWorld.Map.Patches.Count;
+
+            MicrobeSpeciesOrganelleStatistics =
+                new Dictionary<OrganelleDefinition, (double Percentage, double Average)>();
+
+            foreach (var organelle in SimulationParameters.Instance.GetAllOrganelles())
+            {
+                MicrobeSpeciesOrganelleStatistics.Add(organelle, (0, 0));
+            }
+
+            UpdateWorldStatistics();
+        }
+
+        public int TotalSpeciesCount { get; private set; }
+
+        public int CurrentSpeciesCount { get; private set; }
+
+        public int PatchesCount { get; }
+
+        public double PatchSpeciesCountAverage { get; private set; }
+
+        public double PatchSpeciesCountStandardDeviation { get; private set; }
+
+        public long TotalPopulation { get; private set; }
+
+        public double MicrobeSpeciesAverageHexSize { get; private set; }
+
+        public void UpdateWorldStatistics()
+        {
+            TotalSpeciesCount = (int)SpeciesHistoryList.Last().Values.Max(s => s.ID);
```

Comment:
Shouldn't this use "Values.Count" so that this is actually accurate? The IDs aren't absolutely guaranteed to equal the number of species, so I'd avoid making that assumption in any code.

Reply:
The list only contains species that are still alive. Its count can be significantly less than the actual count.

Reply:
But then the maximum ID number is definitely not equal to the "count". Should the variable be perhaps named something else if it doesn't actually have the count in it?

Reply:
That might be true but fundamentally I think the ID should not be *assumed* to be incremental. The code will be much more robust if actual counting is performed here instead of an assumption that might get broken sometime in the future.